packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts

```diff
@@ -607,42 +606,49 @@ export class MLSService extends TypedEventEmitter<Events> {
 
   private async syncKeyPackages() {
     const validKeyPackagesCount = await this.clientValidKeypackagesCount();
-    const minAllowedNumberOfKeyPackages = this.config.nbKeyPackages / 2;
 
-    if (validKeyPackagesCount <= minAllowedNumberOfKeyPackages) {
+    if (validKeyPackagesCount <= this.config.minRequiredNumberOfAvailableKeyPackages) {
       const clientId = this.apiClient.validatedClientId;
-
-      //check numbers of keys on backend
-      const backendKeyPackagesCount = await this.apiClient.api.client.getMLSKeyPackageCount(
-        clientId,
-        numberToHex(this.config.defaultCiphersuite),
-      );
-
-      if (backendKeyPackagesCount <= minAllowedNumberOfKeyPackages) {
-        //upload new keys
-        const newKeyPackages = await this.clientKeypackages(this.config.nbKeyPackages);
-
-        await this.uploadMLSKeyPackages(newKeyPackages, clientId);
-      }
+      await this.uploadMLSKeyPackages(clientId);
     }
   }
 
+  private async getRemoteMLSKeyPackageCount(clientId: string) {
+    return this.apiClient.api.client.getMLSKeyPackageCount(clientId, numberToHex(this.config.defaultCiphersuite));
+  }
+
   /**
-   * Will make the given client mls capable (generate and upload key packages)
+   * Will update the given client on backend with its public key.
    *
    * @param mlsClient Intance of the coreCrypto that represents the mls client
    * @param clientId The id of the client
    */
-  private async uploadMLSPublicKeys(publicKey: Uint8Array, clientId: string) {
+  private async uploadMLSPublicKeys(clientId: string) {
+    const existingPublicKey = await this.apiClient.api.client.getClient(clientId);
```

Comment:
question: Since we have the client at this point https://github.com/wireapp/wire-web-packages/pull/5495/files#diff-0a7e6d4a8ee3b5e3153dbc2336fecd1741ea0f39e7226d6b7a5d63be31b2747fR267
maybe it would be better to inject it, rather than re-fetching (but maybe there is another codepath I'm not seeing)?